Source/Core/Core/IOS/Network/KD/VFF/VFFUtil.cpp

```diff
@@ -237,6 +237,36 @@ static ErrorCode WriteFile(const std::string& filename, const std::vector<u8>& t
   return WC24_OK;
 }
 
+static ErrorCode ReadFile(const std::string& filename, std::vector<u8>& out)
+{
+  FIL src{};
+  const auto open_error_code = f_open(&src, filename.c_str(), FA_READ);
+  if (open_error_code != FR_OK)
+  {
+    ERROR_LOG_FMT(IOS_WC24, "Failed to open file {} in VFF", filename);
+    return WC24_ERR_FILE_OPEN;
+  }
+
+  u32 size = static_cast<u32>(out.size());
+  u32 read_size{};
+  const auto read_error_code = f_read(&src, out.data(), size, &read_size);
+  if (read_error_code != FR_OK)
```

Comment:
Should the two following error cases and also the end of the function `f_close` the opened file handle before returning?

Reply:
They should, I completely missed that.